sphinxgallery/gen_rst.py

```diff
@@ -444,7 +458,12 @@ def generate_file_rst(fname, target_dir, src_dir):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    exec_script = fname.startswith('plot')
+    use_mayavi = gallery_conf.get('use_mayavi', False)
+    if not use_mayavi and any('mayavi' in bl[1] for bl in script_blocks):
```

Comment:
> FWIW I get segfaults from Mayavi, yeah

Sorry not a mayavi user here ... 

I think what I am trying to say is that if you have use_mayavi=False, and have an example that uses mayavi it should still be run and you just wouldn't get the mayavi figures in your example documentation. If the example using mayavi segfaults, then so be it, go and fix your example. To me the use_mayavi flag is more something like find_mayavi_figures than run_scripts_using_mayavi. Maybe it could be renamed to something similar.

There is an argument about having a some skipping functionality in sphinx-gallery but I feel this is orthogonal to this issue. Use cases I know of:
* when you are working on an example, being able to only run this example can speed things up a lot. Full disclosure: I have seen people hitting Control-C on all the examples but the one they cared about to achieve that functionality.
* I think @Eric89GXL mentioned skipping examples tagged as slow which is kind of a more general version of the first bullet point
* skipping examples that uses mayavi could be one, in case you have a mayavi that segfaults at import time and don't have time to fix it.